packages/client/src/DataCommonsClient.ts

```diff
@@ -0,0 +1,480 @@
+const DEFAULT_VARIABLE_PROPS = [NAME_ATTRIBUTE];
+
+export interface DatacommonsWebClientParams {
+  /** Web api root endpoint. Default: `"https://datacommons.org/""` */
+  apiRoot?: string;
+}
+
+export interface BaseGetDataRowsParams {
+  /** Variable DCIDs */
+  variables: string[];
+  /** Example: 2023 */
+  date?: string;
+  facetIds?: string[];
+  /** Fetch these entity properties from the knowledge graph. Default: `["name", "isoCode"]` */
+  entityProps?: string[];
+  /** Fetch these variable properties from the knowledge graph. Default: `["name"]` */
+  variableProps?: string[];
+  /**
+   * Performs per-capita caluclation for any of these variables.
+   * Must be a subset of `variables` param.
+   */
+  perCapitaVariables?: string[];
+}
+
+export interface GetDataRowsParamsWithin extends BaseGetDataRowsParams {
+  /** Parent entity DCID. Example: `"country/USA"` */
+  parentEntity: string;
+  /** Child node type. Example: `"State"` */
+  childType: string;
+}
+
+export interface GetDataRowsParamsEntities extends BaseGetDataRowsParams {
+  /** Entity DCIDs. Example: `["country/USA", "country/IND"]` */
+  entities: string[];
+}
+export type GetDataRowsParams =
+  | GetDataRowsParamsWithin
+  | GetDataRowsParamsEntities;
+
+export type GetGeoJSONParams = GetDataRowsParams & {
+  /** GeoJSON property name in the knowledge graph. Inferred if not provided. */
+  geoJsonProperty?: string;
+};
+
+export interface DataRow {
+  [key: string]: string | number | boolean | null;
+}
+
+export type NodePropValues = Record<string, Record<string, string | null>>;
+
+class DataCommonsWebClient {
+  apiRoot?: string;
+
+  constructor(params?: DatacommonsWebClientParams) {
+    const p = params || {};
+    this.apiRoot =
+      p.apiRoot !== undefined
+        ? p.apiRoot.replace(/\/$/, "")
+        : "https://datacommons.org";
+  }
+
+  async getCsv(params: GetDataRowsParams): Promise<string> {
+    const dataRows = await this.getDataRows(params);
+
+    if (dataRows.length === 0) {
+      return "";
+    }
+
+    const header = Object.keys(dataRows[0]).sort();
+    const rows = dataRows.map((dataRow) =>
+      header.map((column) => dataRow[column])
+    );
+    const csvRows = [header, ...rows];
+    const csvLines = csvRows.map(encodeCsvRow);
+    return csvLines.join("\n");
+  }
+
+  async getGeoJSON(params: GetGeoJSONParams): Promise<FeatureCollection> {
+    const geoJsonProperty = params.geoJsonProperty || "geoJsonCoordinatesDP1";
+    const dataRows = await this.getDataRows({
+      ...params,
+      entityProps: [
+        geoJsonProperty,
+        ...(params.entityProps ? params.entityProps : DEFAULT_ENTITY_PROPS),
+      ],
+    });
+
+    const geoJson: FeatureCollection = {
+      type: "FeatureCollection",
+      features: dataRows
+        .filter((dataRow) => {
+          const geometryString = dataRow[`entity.${geoJsonProperty}`];
+          if (geometryString !== "string") {
```

Comment:
curious what would cause the geometryString to come back as "string". Did you mean to check that the `typeof` geometryString is "string"?

Reply:
Whoops, meant to use `typeof`. Fixed.